From 762254a6184b40ef3e058ba48c4a1b5e01e0541d Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 19 Aug 2025 16:23:23 -0700
Subject: [PATCH 22/24] hipfft and rocfft test integration with fftw-3.3.10

Works for finding the newer fftw3 version that has been
integrated to TheRock build.

Changes in cmake configuration compared to older version:
1) FFTW_INCLUDE_DIRS --> FFTW3_INCLUDE_DIRS
2) FFTW_LIBRARIES
-->FFTW3::fftw3 FFTW3::fftw3f

(fftw3q and fftw3qd may need to be added later if support
 for those are also build to fftw and then added tests
 also)

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 projects/hipfft/clients/tests/CMakeLists.txt |  5 +++--
 projects/rocfft/clients/tests/CMakeLists.txt | 10 ++++++----
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/projects/hipfft/clients/tests/CMakeLists.txt b/projects/hipfft/clients/tests/CMakeLists.txt
index e49c87b6f6..ecbff97486 100644
--- a/projects/hipfft/clients/tests/CMakeLists.txt
+++ b/projects/hipfft/clients/tests/CMakeLists.txt
@@ -158,7 +158,7 @@ foreach( target ${TEST_TARGETS} )
   target_include_directories( ${target}
     PRIVATE
     $<BUILD_INTERFACE:${Boost_INCLUDE_DIRS}>
-    $<BUILD_INTERFACE:${FFTW_INCLUDE_DIRS}>
     $<BUILD_INTERFACE:${hip_INCLUDE_DIRS}>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../library/include>
   )
@@ -166,7 +166,8 @@ foreach( target ${TEST_TARGETS} )
   target_link_libraries( ${target}
     PRIVATE
     hip::hipfft
-    ${FFTW_LIBRARIES}
+    FFTW3::fftw3
+    FFTW3::fftw3f
   )
 
   if( HIPFFT_MPI_ENABLE )
diff --git a/projects/rocfft/clients/tests/CMakeLists.txt b/projects/rocfft/clients/tests/CMakeLists.txt
index cfec9a283d..81282b237e 100644
--- a/projects/rocfft/clients/tests/CMakeLists.txt
+++ b/projects/rocfft/clients/tests/CMakeLists.txt
@@ -175,7 +175,7 @@ if( BUILD_FFTW OR NOT FFTW_FOUND )
     COMPONENT clients-common
   )
 else()
-  include_directories(${FFTW_INCLUDE_DIRS})
+  include_directories(${FFTW3_INCLUDE_DIRS})
 endif()
 
 set( rocfft-test_include_dirs
@@ -186,7 +186,7 @@ set( rocfft-test_include_dirs
   )
 
 set( rocfft-test_link_libs
-  ${FFTW_LIBRARIES} OpenMP::OpenMP_CXX
+  FFTW3::fftw3 FFTW3::fftw3f OpenMP::OpenMP_CXX
   )
 
 include( ../cmake/build-gtest.cmake )
@@ -353,7 +353,8 @@ if( ROCFFT_MPI_ENABLE )
       hip::hiprand
       hip::device
       MPI::MPI_CXX
-      ${FFTW_LIBRARIES}
+      FFTW3::fftw3
+      FFTW3::fftw3f
       "mpi_gtl_hsa"
     )
       get_filename_component( MPI_LIBDIR ${MPI_LIBRARY} DIRECTORY )
@@ -366,7 +367,8 @@ if( ROCFFT_MPI_ENABLE )
       hip::hiprand
       hip::device
       MPI::MPI_CXX
-      ${FFTW_LIBRARIES}
+      FFTW3::fftw3
+      FFTW3::fftw3f
     )    
     endif()
     set_target_properties(${worker}
-- 
2.34.1
